superscore/widgets/page/entry.py

```diff
@@ -101,6 +106,41 @@ def closeEvent(self, a0: QCloseEvent) -> None:
         self.sub_pv_table_view._model.stop_polling(wait_time=5000)
         return super().closeEvent(a0)
 
+    def take_snapshot(self) -> Snapshot:
```

Comment:
```suggestion
    def take_snapshot(self) -> Optional[Snapshot]:
```

If the snapshot doesn't have an origin collection this just returns None.

The choice to throw Exceptions in `Client.find_origin_collection` and return `None` here still seems a little at odds, I feel like we should choose one approach.  Or maybe I'm still hung up on the fact that we throw exceptions when an optional field is `None`

Reply:
I don't see much of a difference between returning `None` and raising an error in the lower method. In both cases, the caller just has to implement a check, either an `is None` or an `except`.  `find_origin_collection` can propagate `EntryNotFoundError`s from `backend.get_entry` anyway, so using a `ValueError` just made the handling cleaner.

Since I attach `take_snapshot` to a button, I don't know how to handle raised exceptions, which is why I return `None`.

Reply:
Yeah I'll put down the issue of none vs exceptions, but I do think the type hint should be updated.

Sorry for being annoying about it 😓

Reply:
Right, I'll still update the type hint.

My reasoning for using an error is that in my mind `find_origin_collection` is supposed to return the `Collection` entry associated with a nestable, and if it can't find such an entry, then that's an error.  Sure `Snapshot.origin_collection` is optional, but if it's `None` then `find_origin_collection` can't do its job because there's no entry to fetch.

But like I said, I don't actually see much pragmatic difference, so I don't mind changing it if it's bothersome.

Reply:
I think your rationale makes sense, the only difference in my mind being that `find_origin_collection` returning `None` still seems to me like it's doing the job, there's just no origin to return.

No need to change it.  If it's a problem in the future we'll find out and fix it then.